Replace (most of) card package with auto-generated code

dclaux opened the pull request:
## Overview
This PR replaces most of the card package code with auto-generated code that is 100% accurately represents the Adaptive Card specification. The auto-generated code is simply copy/pasted from [this tool](https://1js-static-pages.azurewebsites.net/refs/heads/user/dclaux/ac-typescript-code-generator-update/ac-react-schema-tool/index.html?topic=ac-typescript-proxy)

The auto-generated code presents the following differences compared to the code it replaces:
- Some type names are different
- Enum values start with a capital letter (this can be changed if really necessary, we can discuss)
- There is now one `isXXX` method for each type (there used to only be `isCard`)
- For now at least the entire auto-generated code lives in one single file. We can always look into splitting it into multiple files in the future.

Teams-specific action stuff remains manually written although now it extends classes from the auto-generated code.

## Updating the `teams.cards` package going forward
- Navigate to the [TypeScript interface generator tool](https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy). The content on that page is dynamically generate every time the page is visited. It 100% matches what is documented, guarantying accuracy and completeness.
- De-select target apps other than Teams
- Make sure the "Generate builder classes" option is selected (it is by default)
- Copy the entire generated code
- Replace the entirety of the `src/core.ts` file with that copied content

The tool cannot currently be automated - there's no CLI available for it.

heyitsaamir commented:
This looks reasonable to me!
1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
  - withBody -> constructor of `AdaptiveCard`
  - addAction -> withAction
  - pascal casing changes to string literals
 
Did you see anything else?

2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator

dclaux commented:
> This looks reasonable to me!
> 
> 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> 
> * withBody -> constructor of `AdaptiveCard`
> * addAction -> withAction
> * pascal casing changes to string literals
> 
> Did you see anything else?
> 
> 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator

The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.

heyitsaamir commented:
> > This looks reasonable to me!
> > 
> > 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> > 
> > 
> > * withBody -> constructor of `AdaptiveCard`
> > * addAction -> withAction
> > * pascal casing changes to string literals
> > 
> > Did you see anything else?
> > 
> > 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator
> 
> The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.

Hm, isn't that just the output? The top of the file says:
```
This file was automatically generated by a tool on 05/02/2025, 5:09 PM UTC.
It includes declarations for Adaptive Card features available in Teams, Copilot, Outlook, Word, Excel, PowerPoint.
*/
```

dclaux commented:
> > > This looks reasonable to me!
> > > 
> > > 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> > > 
> > > 
> > > * withBody -> constructor of `AdaptiveCard`
> > > * addAction -> withAction
> > > * pascal casing changes to string literals
> > > 
> > > Did you see anything else?
> > > 
> > > 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator
> > 
> > 
> > The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.
> 
> Hm, isn't that just the output? The top of the file says:
> 
> ```
> This file was automatically generated by a tool on 05/02/2025, 5:09 PM UTC.
> It includes declarations for Adaptive Card features available in Teams, Copilot, Outlook, Word, Excel, PowerPoint.
> */
> ```

The tool cannot currently be automated. It is built into that UI. In the future, you will have to navigate to that page, copy the generated content (after making sure you de-select target apps other than Teams) and fully replace the content of `src/core.ts` with it.

aacebo reviewed (changes requested):
one missing cli template update https://github.com/microsoft/teams.ts/blob/main/packages/cli/templates/typescript/graph/src/index.ts#L25

corinagum reviewed (dismissed):
Thanks for the fast turn around David! Looks like there is a build failure on `src/activities/utils/to-activity-params.spec.ts`, but other than that LGTM!

heyitsaamir commented:
> > > > This looks reasonable to me!
> > > > 
> > > > 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> > > > 
> > > > 
> > > > * withBody -> constructor of `AdaptiveCard`
> > > > * addAction -> withAction
> > > > * pascal casing changes to string literals
> > > > 
> > > > Did you see anything else?
> > > > 
> > > > 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator
> > > 
> > > 
> > > The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.
> > 
> > 
> > Hm, isn't that just the output? The top of the file says:
> > ```
> > This file was automatically generated by a tool on 05/02/2025, 5:09 PM UTC.
> > It includes declarations for Adaptive Card features available in Teams, Copilot, Outlook, Word, Excel, PowerPoint.
> > */
> > ```
> 
> The content is dynamically generated every time you visit that page. The tool cannot currently be automated. It is built into that UI. In the future, you will have to navigate to that page, copy the generated content (after making sure you de-select target apps other than Teams) and fully replace the content of `src/core.ts` with it.

Ah gotcha. I'm actually asking where the JS code that runs on the webapp lives. If there's a bug, how do we fix it? Do we just message you? Or can someone go out, fix the generator?

ryanbliss reviewed (dismissed):
LGTM, seems comprehensive.

dclaux commented:
> one missing cli template update https://github.com/microsoft/teams.ts/blob/main/packages/cli/templates/typescript/graph/src/index.ts#L25

Fixed. Also fixed a snippet that still referred to `ICard`

dclaux commented on tests/dialogs/src/index.ts:
```diff
@@ -157,40 +158,39 @@ app.on('dialog.open', async ({ activity, next }) => {
-          .withId('name')
-          .withPlaceholder('Enter your name')
-      )
+    const dialogCard = new AdaptiveCard(
```
Resolving this, I suppose if someone disagrees they can reopen and we can have a chat.

dclaux commented:
> > > > > This looks reasonable to me!
> > > > > 
> > > > > 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> > > > > 
> > > > > 
> > > > > * withBody -> constructor of `AdaptiveCard`
> > > > > * addAction -> withAction
> > > > > * pascal casing changes to string literals
> > > > > 
> > > > > Did you see anything else?
> > > > > 
> > > > > 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator
> > > > 
> > > > 
> > > > The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.
> > > 
> > > 
> > > Hm, isn't that just the output? The top of the file says:
> > > ```
> > > This file was automatically generated by a tool on 05/02/2025, 5:09 PM UTC.
> > > It includes declarations for Adaptive Card features available in Teams, Copilot, Outlook, Word, Excel, PowerPoint.
> > > */
> > > ```
> > 
> > 
> > The content is dynamically generated every time you visit that page. The tool cannot currently be automated. It is built into that UI. In the future, you will have to navigate to that page, copy the generated content (after making sure you de-select target apps other than Teams) and fully replace the content of `src/core.ts` with it.
> 
> Ah gotcha. I'm actually asking where the JS code that runs on the webapp lives. If there's a bug, how do we fix it? Do we just message you? Or can someone go out, fix the generator?

It's all in the 1JS repo. Namely, here: https://office.visualstudio.com/Office/_git/1JS?path=/midgard/packages/ac-react-schema-tool

dclaux commented:
> Thanks for the fast turn around David! Looks like there is a build failure on `src/activities/utils/to-activity-params.spec.ts`, but other than that LGTM!

Indeed. And there were actually a couple other missed references as well in other files. They should all be fixed now.

dclaux commented:
@aacebo I am still seeing linting errors despite running `lint:fix`. Am I missing something?

dclaux commented:
> > > > > This looks reasonable to me!
> > > > > 
> > > > > 1. I'll test out the cards sample out tomorrow, but it's great to see that there were minor breaking changes. From what I saw:
> > > > > 
> > > > > 
> > > > > * withBody -> constructor of `AdaptiveCard`
> > > > > * addAction -> withAction
> > > > > * pascal casing changes to string literals
> > > > > 
> > > > > Did you see anything else?
> > > > > 
> > > > > 2. Where is this _tool_ you mentioned? It'd be good to know how we can generate these if you're not available, or if we need to debug issues in the generator
> > > > 
> > > > 
> > > > The permanent internal tool is at https://1js-static-pages-release.azurewebsites.net/azurestaticapps/ac-react-schema-tool/index.html?topic=ac-typescript-proxy (bookmark it). Right now it is not up to date as I've been working on this PR alongside another for the tool and will not merge the tool code until I know we're good here. But there is a test link associated with my other PR in the description above.
> > > 
> > > 
> > > Hm, isn't that just the output? The top of the file says:
> > > ```
> > > This file was automatically generated by a tool on 05/02/2025, 5:09 PM UTC.
> > > It includes declarations for Adaptive Card features available in Teams, Copilot, Outlook, Word, Excel, PowerPoint.
> > > */
> > > ```
> > 
> > 
> > The content is dynamically generated every time you visit that page. The tool cannot currently be automated. It is built into that UI. In the future, you will have to navigate to that page, copy the generated content (after making sure you de-select target apps other than Teams) and fully replace the content of `src/core.ts` with it.
> 
> Ah gotcha. I'm actually asking where the JS code that runs on the webapp lives. If there's a bug, how do we fix it? Do we just message you? Or can someone go out, fix the generator?

Forgot to answer your other question - if there are issues with the generator, you can ping me and I'll make it a point to address them quickly, but you're also welcome to contribute to the AC stack in 1JS if that's what you prefer.
